community/FfDL-Seldon/pytorch-model/README.md

```diff
@@ -0,0 +1,49 @@
+# TensorFlow MNIST Classifier
+
+# Train Model
+
+Train the [pyTorch MNIST model](https://github.com/IBM/FfDL/tree/master/etc/examples/pytorch-model) following the steps in the [user guide](https://github.com/IBM/FfDL#6-detailed-testing-instructions).
+
+# Wrap the Runtime Scorer
+You can skip this step if you are happy to use the already packaged image ```seldonio/ffdl-pymnist``` from DockerHub.
+
+The runtime MNIST scrorer is contained within a standalone [python class PyMnist.py](./PyMnist.py). This needs to be packaged in a Docker container to run within Seldon. For this we use [Redhat's Source-to-image](https://github.com/openshift/source-to-image).
+
+ * Install [S2I](https://github.com/openshift/source-to-image#installation)
+ * From the tf-model folder run, (*change seldonio to your Docker repo*):
+```
+s2i build . seldonio/seldon-core-s2i-python2 seldonio/ffdl-pymnist:0.1
```

Comment:
On Mac, using s2i with `seldonio/seldon-core-s2i-python2` gives me the following errors.
```shell
$ s2i build . seldonio/seldon-core-s2i-python2 seldonio/ffdl-pymnist:0.1
---> Installing application source...
---> Installing dependencies ...
Collecting torch==0.4.0 (from -r requirements.txt (line 1))
Downloading https://files.pythonhosted.org/packages/df/a4/7f5ec6e9df1bf13f1881353702aa9713fcd997481b26018f35e0be85faf7/torch-0.4.0-cp27-cp27mu-manylinux1_x86_64.whl (484.0MB)
/s2i/bin/assemble: line 59:    12 Killed                  pip install -r requirements.txt
Build failed
ERROR: An error occurred: non-zero (13) exit code from seldonio/seldon-core-s2i-python2
```

However, it works fine with `seldonio/seldon-core-s2i-python3`. It's an issue with python2 on Mac?

Reply:
Could it be related to https://github.com/openshift/source-to-image/issues/864
proxy/https issue?

Reply:
After some investigations, I found out this is a out of memory error when doing the pip install. For some reason, when using Python 2.7's pip 9.0.1 to install the torch manylinux package will need more than 6GB of RAM for etag caching. 

So to fix this we can either:
1. Upgrade pip to the latest version for the `seldonio/seldon-core-s2i-python2` image or
2. Build with `seldonio/seldon-core-s2i-python3` or
3. Ask users to give 8GB or more memory to their Docker engine.

Reply:
Option 1 or 3 seems easiest. I'm not sure later versions of pip will fix this? We are using python2 or python3 base images. Are you running Docker on a machine with less than 8G?

Reply:
I think we can just go with Option 1, it's was a problem to me because I limit my docker engine to only consume up to 4GB of Memory.

Reply:
Let`s add that as a disclaimer in the writeup - min 8GB mem required